sklearnDev CI tests failing on sklearn.preprocessing.label module

Here's a traceback from #528 

``` python
______________ ERROR collecting tests/preprocessing/test_label.py ______________
tests/preprocessing/test_label.py:10: in <module>
    import dask_ml.preprocessing as dpp
dask_ml/preprocessing/__init__.py:3: in <module>
    from ._encoders import OneHotEncoder
dask_ml/preprocessing/_encoders.py:10: in <module>
    from .label import _encode, _encode_dask_array
dask_ml/preprocessing/label.py:10: in <module>
    from sklearn.preprocessing import label as sklabel
/usr/share/miniconda/envs/dask-ml-t…
/usr/share/miniconda/envs/dask-ml-test/lib/python3.7/site-packages/sklearn/utils/deprecation.py:143: in _raise_dep_warning_if_not_pytest
    warnings.warn(message, DeprecationWarning)
E   DeprecationWarning: The sklearn.preprocessing.label module is  deprecated in version 0.22
 and will be removed in version 0.24. The corresponding classes / functions should instead be
 imported from sklearn.preprocessing. Anything that cannot be imported from
 sklearn.preprocessing is now part of the private API.
```

--- a/dask_ml/_compat.py
+++ b/dask_ml/_compat.py
@@ -35,3 +35,12 @@ def check_is_fitted(est, attributes=None):
         args = (attributes,)
 
     return sklearn.utils.validation.check_is_fitted(est, *args)
+
+
+def _check_multimetric_scoring(estimator, scoring=None):
+    if SK_022:
+        from sklearn.metrics._scorer import _check_multimetric_scoring
+    else:
+        from sklearn.metrics._scorer import _check_multimetric_scoring
+
+    return _check_multimetric_scoring(estimator, scoring)

--- a/dask_ml/_compat.py
+++ b/dask_ml/_compat.py
@@ -41,6 +41,6 @@ def _check_multimetric_scoring(estimator, scoring=None):
     if SK_022:
         from sklearn.metrics._scorer import _check_multimetric_scoring
     else:
-        from sklearn.metrics._scorer import _check_multimetric_scoring
+        from sklearn.metrics.scorer import _check_multimetric_scoring
 
     return _check_multimetric_scoring(estimator, scoring)

--- a/dask_ml/linear_model/stochastic_gradient.py
+++ b/dask_ml/linear_model/stochastic_gradient.py
@@ -1,15 +1,18 @@
-from sklearn.linear_model import stochastic_gradient as _sg
+import sklearn.linear_model
 
 from .._partial import _BigPartialFitMixin, _copy_partial_doc
 
 
 @_copy_partial_doc
-class PartialSGDClassifier(_BigPartialFitMixin, _sg.SGDClassifier):
+class PartialSGDClassifier(_BigPartialFitMixin, sklearn.linear_model.SGDClassifier):
 
     _init_kwargs = ["classes"]
     _fit_kwargs = ["classes"]
 
 
 @_copy_partial_doc
-class PartialSGDRegressor(_BigPartialFitMixin, _sg.SGDRegressor):
+class PartialSGDRegressor(_BigPartialFitMixin, sklearn.linear_model.SGDRegressor):
     pass
+
+
+__all__ = ["PartialSGDClassifier", "PartialSGDRegressor"]

--- a/dask_ml/metrics/scorer.py
+++ b/dask_ml/metrics/scorer.py
@@ -1,5 +1,4 @@
-from sklearn.metrics import make_scorer
-from sklearn.metrics.scorer import check_scoring as sklearn_check_scoring
+from sklearn.metrics import check_scoring as sklearn_check_scoring, make_scorer
 
 from . import accuracy_score, log_loss, mean_squared_error, r2_score
 

--- a/dask_ml/model_selection/_incremental.py
+++ b/dask_ml/model_selection/_incremental.py
@@ -13,7 +13,7 @@ import toolz
 from dask.distributed import Future, default_client, futures_of, wait
 from distributed.utils import log_errors
 from sklearn.base import clone
-from sklearn.metrics.scorer import check_scoring
+from sklearn.metrics import check_scoring
 from sklearn.model_selection import ParameterGrid, ParameterSampler
 from sklearn.utils import check_random_state
 from sklearn.utils.metaestimators import if_delegate_has_method

--- a/dask_ml/model_selection/_search.py
+++ b/dask_ml/model_selection/_search.py
@@ -1183,7 +1183,7 @@ class DaskBaseSearchCV(BaseEstimator, MetaEstimatorMixin):
             Parameters passed to the ``fit`` method of the estimator
         """
         estimator = self.estimator
-        from sklearn.metrics.scorer import _check_multimetric_scoring
+        from .._compat import _check_multimetric_scoring
 
         scorer, multimetric = _check_multimetric_scoring(
             estimator, scoring=self.scoring

--- a/dask_ml/preprocessing/data.py
+++ b/dask_ml/preprocessing/data.py
@@ -8,11 +8,11 @@ import dask.array as da
 import dask.dataframe as dd
 import numpy as np
 import pandas as pd
+import sklearn.preprocessing
 from dask import compute
 from pandas.api.types import is_categorical_dtype
 from scipy import stats
 from sklearn.base import BaseEstimator, TransformerMixin
-from sklearn.preprocessing import data as skdata
 from sklearn.utils.validation import check_random_state
 
 from dask_ml._compat import DASK_110, SK_022, blockwise, check_is_fitted
@@ -21,11 +21,30 @@ from dask_ml.utils import check_array, handle_zeros_in_scale
 
 _PANDAS_VERSION = LooseVersion(pd.__version__)
 _HAS_CTD = _PANDAS_VERSION >= "0.21.0"
+BOUNDS_THRESHOLD = 1e-7
 
 
-class StandardScaler(skdata.StandardScaler):
+def _handle_zeros_in_scale(scale, copy=True):
+    """ Makes sure that whenever scale is zero, we handle it correctly.
 
-    __doc__ = skdata.StandardScaler.__doc__
+    This happens in most scalers when we have constant features."""
+
+    # if we are fitting on 1D arrays, scale might be a scalar
+    if np.isscalar(scale):
+        if scale == 0.0:
+            scale = 1.0
+        return scale
+    elif isinstance(scale, np.ndarray):
+        if copy:
+            # New array to avoid side-effects
+            scale = scale.copy()
+        scale[scale == 0.0] = 1.0
+        return scale
+
+
+class StandardScaler(sklearn.preprocessing.StandardScaler):
+
+    __doc__ = sklearn.preprocessing.StandardScaler.__doc__
 
     def fit(self, X, y=None):
         self._reset()
@@ -68,9 +87,9 @@ class StandardScaler(skdata.StandardScaler):
         return X
 
 
-class MinMaxScaler(skdata.MinMaxScaler):
+class MinMaxScaler(sklearn.preprocessing.MinMaxScaler):
 
-    __doc__ = skdata.MinMaxScaler.__doc__
+    __doc__ = sklearn.preprocessing.MinMaxScaler.__doc__
 
     def fit(self, X, y=None):
         self._reset()
@@ -131,9 +150,9 @@ class MinMaxScaler(skdata.MinMaxScaler):
         return X
 
 
-class RobustScaler(skdata.RobustScaler):
+class RobustScaler(sklearn.preprocessing.RobustScaler):
 
-    __doc__ = skdata.RobustScaler.__doc__
+    __doc__ = sklearn.preprocessing.RobustScaler.__doc__
 
     def _check_array(self, X, *args, **kwargs):
         X = check_array(X, accept_dask_dataframe=True, **kwargs)
@@ -162,7 +181,7 @@ class RobustScaler(skdata.RobustScaler):
         quantiles = da.vstack(quantiles).compute()
         self.center_ = quantiles[:, 1]
         self.scale_ = quantiles[:, 2] - quantiles[:, 0]
-        self.scale_ = skdata._handle_zeros_in_scale(self.scale_, copy=False)
+        self.scale_ = _handle_zeros_in_scale(self.scale_, copy=False)
         return self
 
     def transform(self, X):
@@ -223,14 +242,16 @@ class RobustScaler(skdata.RobustScaler):
         return X
 
 
-class QuantileTransformer(skdata.QuantileTransformer):
+class QuantileTransformer(sklearn.preprocessing.QuantileTransformer):
     """Transforms features using quantile information.
 
     This implementation differs from the scikit-learn implementation
     by using approximate quantiles. The scikit-learn docstring follows.
     """
 
-    __doc__ = __doc__ + "\n".join(skdata.QuantileTransformer.__doc__.split("\n")[1:])
+    __doc__ = __doc__ + "\n".join(
+        sklearn.preprocessing.QuantileTransformer.__doc__.split("\n")[1:]
+    )
 
     def _check_inputs(self, X, accept_sparse_negative=False, copy=False):
         kwargs = {}
@@ -294,8 +315,8 @@ class QuantileTransformer(skdata.QuantileTransformer):
                 # else output distribution is already a uniform distribution
 
         if output_distribution == "normal":
-            lower_bounds_idx = X_col - skdata.BOUNDS_THRESHOLD < lower_bound_x
-            upper_bounds_idx = X_col + skdata.BOUNDS_THRESHOLD > upper_bound_x
+            lower_bounds_idx = X_col - BOUNDS_THRESHOLD < lower_bound_x
+            upper_bounds_idx = X_col + BOUNDS_THRESHOLD > upper_bound_x
         if output_distribution == "uniform":
             lower_bounds_idx = X_col == lower_bound_x
             upper_bounds_idx = X_col == upper_bound_x
@@ -322,8 +343,8 @@ class QuantileTransformer(skdata.QuantileTransformer):
                 # find the value to clip the data to avoid mapping to
                 # infinity. Clip such that the inverse transform will be
                 # consistent
-                clip_min = stats.norm.ppf(skdata.BOUNDS_THRESHOLD - np.spacing(1))
-                clip_max = stats.norm.ppf(1 - (skdata.BOUNDS_THRESHOLD - np.spacing(1)))
+                clip_min = stats.norm.ppf(BOUNDS_THRESHOLD - np.spacing(1))
+                clip_max = stats.norm.ppf(1 - (BOUNDS_THRESHOLD - np.spacing(1)))
                 X_col = da.clip(X_col, clip_min, clip_max)
 
             # else output distribution is uniform and the ppf is the
@@ -941,7 +962,7 @@ class OrdinalEncoder(BaseEstimator, TransformerMixin):
         return X
 
 
-class PolynomialFeatures(skdata.PolynomialFeatures):
+class PolynomialFeatures(sklearn.preprocessing.PolynomialFeatures):
     """    preserve_dataframe : boolean
             If True, preserve pandas and dask dataframes after transforming.
             Using False (default) returns numpy or dask arrays and mimics
@@ -950,7 +971,9 @@ class PolynomialFeatures(skdata.PolynomialFeatures):
         Examples
     """
 
-    splitted_orig_doc = skdata.PolynomialFeatures.__doc__.split("    Examples\n")
+    splitted_orig_doc = sklearn.preprocessing.PolynomialFeatures.__doc__.split(
+        "    Examples\n"
+    )
     __doc__ = "".join([splitted_orig_doc[0], __doc__, splitted_orig_doc[1]])
 
     def __init__(
@@ -964,7 +987,7 @@ class PolynomialFeatures(skdata.PolynomialFeatures):
         self.preserve_dataframe = preserve_dataframe
 
     def fit(self, X, y=None):
-        self._transformer = skdata.PolynomialFeatures(
+        self._transformer = sklearn.preprocessing.PolynomialFeatures(
             degree=self.degree,
             interaction_only=self.interaction_only,
             include_bias=self.include_bias,

--- a/dask_ml/preprocessing/label.py
+++ b/dask_ml/preprocessing/label.py
@@ -7,12 +7,12 @@ import dask.dataframe as dd
 import numpy as np
 import pandas as pd
 import scipy.sparse
-from sklearn.preprocessing import label as sklabel
+import sklearn.preprocessing
 
 from .._compat import check_is_fitted
 
 
-class LabelEncoder(sklabel.LabelEncoder):
+class LabelEncoder(sklearn.preprocessing.LabelEncoder):
     """Encode labels with value between 0 and n_classes-1.
 
     .. note::

--- a/tests/model_selection/dask_searchcv/test_model_selection.py
+++ b/tests/model_selection/dask_searchcv/test_model_selection.py
@@ -142,8 +142,8 @@ def test_kfolds(cls, has_shuffle):
             cls(shuffle=True, random_state=2, n_splits=3)
         )
 
-        assert tokenize(cls(shuffle=False, random_state=0, n_splits=3)) == tokenize(
-            cls(shuffle=False, random_state=2, n_splits=3)
+        assert tokenize(cls(shuffle=False, random_state=None, n_splits=3)) == tokenize(
+            cls(shuffle=False, random_state=None, n_splits=3)
         )
 
     cv = cls(n_splits=3)

--- a/tests/model_selection/dask_searchcv/test_model_selection.py
+++ b/tests/model_selection/dask_searchcv/test_model_selection.py
@@ -21,7 +21,6 @@ from sklearn.decomposition import PCA
 from sklearn.ensemble import RandomForestClassifier
 from sklearn.exceptions import FitFailedWarning, NotFittedError
 from sklearn.feature_selection import SelectKBest
-from sklearn.metrics.scorer import _passthrough_scorer
 from sklearn.model_selection import (
     GridSearchCV,
     GroupKFold,
@@ -60,6 +59,11 @@ if SK_022:
     iid = {}
 else:
     iid = {"iid": True}
+
+
+def _passthrough_scorer(estimator, *args, **kwargs):
+    """Function that wraps estimator.score"""
+    return estimator.score(*args, **kwargs)
 
 
 class assert_dask_compute(Callback):

--- a/tests/model_selection/dask_searchcv/test_model_selection_sklearn.py
+++ b/tests/model_selection/dask_searchcv/test_model_selection_sklearn.py
@@ -978,7 +978,7 @@ def test_search_cv_results_none_param():
     X, y = [[1], [2], [3], [4], [5]], [0, 0, 0, 0, 1]
     estimators = (DecisionTreeRegressor(), DecisionTreeClassifier())
     est_parameters = {"random_state": [0, None]}
-    cv = KFold(random_state=0, n_splits=2)
+    cv = KFold(random_state=0, n_splits=2, shuffle=True)
 
     for est in estimators:
         grid_search = dcv.GridSearchCV(est, est_parameters, cv=cv).fit(X, y)
@@ -1053,7 +1053,7 @@ def test_grid_search_with_multioutput_data():
     X, y = make_multilabel_classification(return_indicator=True, random_state=0)
 
     est_parameters = {"max_depth": [1, 2, 3, 4]}
-    cv = KFold(random_state=0, n_splits=3)
+    cv = KFold(random_state=0, n_splits=3, shuffle=True)
 
     estimators = [
         DecisionTreeRegressor(random_state=0),

